src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
```

Comment:
Hi, not exactly a coding violation but I was thinking to append 'Task' behind the class names for the types of Tasks?
```suggestion
public class DeadlineTask extends Task {
```
I would say this could be more useful for classes that potentially have multiple meanings -> Event could refer to button onClick event etc. People who view your code base can also easily navigate to whichever task type they want to review easily.

Let me know what you think! :)